### PR TITLE
security fixes npm part

### PR DESCRIPTION
**Pull request by ivolzhevbt**
Hi maintainers! I'm vscode-kubernetes-tools and I work at [Bison Trails](https://bisontrails.co/). We're doing an internal hackathon this week.

Our team loves vscode-kubernetes-tools, but it got flagged during a recent security review. Our team voted to use our hackathon time to patch vulnerabilities in our favorite OSS tools and specifically to try to get vscode-kubernetes-tools approved for usage within our organization. 

## What's here:
- updated package version which are free from vulnerability
- dependencies forced to vulnerability-free versions when intermediate dependencies are not patched

## Impact: 
These changes should improve vscode-kubernetes-tools's security posture. We identified these vulnerabilities using [Salus](https://github.com/coinbase/salus) and the [OSSF scorecard](https://github.com/ossf/scorecard). The specific issues I am hoping to address here are:

| advisory | package | vulnerability | severity |
| --- | --- | --- | --- |
| [565](https://npmjs.com/advisories/565) | ssri | Regex DoS | mod |
| [1179](https://npmjs.com/advisories/1179) | minimist | Prototype Pollution | low | 
| [1500](https://npmjs.com/advisories/1500) | yargs-parser | Prototype Pollution | low |  
| [1589](https://npmjs.com/advisories/1589) | ini | Prototype Pollution | low  |
| [1654](https://npmjs.com/advisories/1654) | y18n | Prototype Pollution | high | 
| [1673](https://npmjs.com/advisories/1673) | lodash | Command Injectition | high | 
| [1674](https://npmjs.com/advisories/1674) | underscore | Arbitrary Code execution | high |
| [1677](https://npmjs.com/advisories/1677) | hosted-git-info | Regex DoS | mod |
| [1678](https://npmjs.com/advisories/1678) | url-parse | Path traversal | high |
| [1748](https://npmjs.com/advisories/1748) | ws | Regex DoS | mod |
| [1751](https://npmjs.com/advisories/1751) | glob-parent | Regex DoS | mod |
| [1754](https://npmjs.com/advisories/1754) | css-what | DoS | high |
| [1755](https://npmjs.com/advisories/1755) | normalize-url | Regex DoS | high | 

Thank you so much for your work maintaining vscode-kubernetes-tools, and thank you for taking the time to look at these proposed changes!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1217,7 +1217,8 @@
         "compile:view": "webpack --mode none --config src/components/logs/webpack.config.js",
         "watch": "webpack --mode development --watch --info-verbosity verbose",
         "test-compile": "tsc -p ./",
-        "test": "npm run test-compile && node ./out/test/runTest.js"
+        "test": "npm run test-compile && node ./out/test/runTest.js",
+        "preinstall": "npx npm-force-resolutions"
```
Comment (brendandburns) [inline package.json:1221]: which dependency requires this?

Comment (ivolzhevbt) [reply package.json:1221]: npx is used to force every dependency in resolutions:
"resolutions": {
        "minimist": "1.2.5",
        "yargs-parser": "13.1.2",
        "ini": "1.3.8",
        "glob-parent": "5.1.2",
        "normalize-url": "4.5.1"
}
offending versions are included through following dependencies:
* minimist: gulp -> glob-watchers -> chokidar -> fsevents -> node-pre-gyp -> (mkdirp, rc) -> minimist
*  yargs-parser: gulp -> gulp-cli -> yargs -> yargs-parser
* ini: gulp -> glob-watchers -> chokidar -> fsevents -> node-pre-gyp -> rc -> ini
* glob-parent: gulp > vinyl-fs > glob-stream > glob-parent, gulp > glob-watcher > chokidar > glob-parent
* normalize-url: download -> got -> cacheable-request -> normalize-url

**Comment by itowlson** [comment]
@ivolzhevbt Thanks heaps to you and the other Bison Trails hackathon participants for helping to address these dependencies - it's a chore and we really appreciate it!

**Review by itowlson — APPROVED** [review]
Looks good - and thanks for introducing us to the npm-force-resolutions trick. I bet that comes in handy again in future.

**Comment by itowlson** [comment]
Hmm, looks like this causes `npm list` to find extraneous packages (included in `node_modules` under other dependencies such as `tar`) and to error.  Which is a problem because `vsce` calls that to build the marketplace package...

**Comment by ivolzhevbt** [comment]
@itowlson oh
i'll need some time to investigate

at the moment though vsce is not working and it is not found on npmjs site for some reason

**Comment by itowlson** [comment]
It's weird - and definitely not your fault.  `npm list` is so pernickety and so buggy and yet `vsce` is so insistent on trusting it.

It probably confuses things yet further that we have webpack in the mix as well.  Which really ought to be doing the tree shaking so `vsce` doesn't have to care, but _somehow it never seems to work that way_.

**Comment by itowlson** [comment]
It looks like a lot of the modules that NPM was flagging as having vulnerabilities are used only in the dev environment, notably by `gulp`.  The rest can be addressed by updating the things you identified, plus the Kubernetes client library and `got`.  Then `npm audit --production` shows no remaining vulnerabilities.

@ivolzhevbt If we make these updates, so that `npm audit --production` shows no vulnerabilities, but remove the `npm-force-resolutions`, so that `vsce` doesn't complain... will that meet your security needs?

**Pull request merged**